driver-core/src/main/com/mongodb/connection/CommandMessage.java

```diff
@@ -68,6 +70,31 @@
         this.readPreference = readPreference;
     }
 
+    BsonDocument getCommandDocument(final ByteBufferBsonOutput bsonOutput) {
+        ByteBufBsonDocument byteBufBsonDocument = ByteBufBsonDocument.createOne(bsonOutput,
+                getEncodingMetadata().getFirstDocumentPosition());
+        BsonDocument commandBsonDocument;
+
+        if (useOpMsg() && containsPayload()) {
+            commandBsonDocument = byteBufBsonDocument.toBsonDocument();
+
+            int payloadStartPosition = getEncodingMetadata().getFirstDocumentPosition()
+                    + byteBufBsonDocument.getSizeInBytes()
+                    + 1 // payload type
+                    + 4 // payload size
+                    + payload.getPayloadName().getBytes(Charset.forName("UTF-8")).length + 1;  // null-terminated UTF-8 payload name
+            commandBsonDocument.append(payload.getPayloadName(),
+                    new BsonArray(ByteBufBsonDocument.create(bsonOutput, payloadStartPosition)));
```

Comment:
As `ByteBufBsonDocument.create` and `ByteBufBsonDocument.createOne` are poorly named as create actually creates a list. This made me double take at whats added to the BsonArray. 

Can we take this opportunity to rename?

Reply:
createMany or createList?

Reply:
createList seems most inkeeping with existing API names